examples/dump_cfi.rs

```diff
@@ -18,7 +18,8 @@ fn print_error(error: &Error) {
 fn dump_cfi<P: AsRef<Path>>(path: P) -> Result<(), Error> {
     let path = path.as_ref();
 
-    let buffer = ByteView::open(path)?;
+    let dsym_path = path.resolve_dsym();
+    let buffer = ByteView::open(dsym_path.as_deref().unwrap_or(path))?;
```

Comment:
Is this `.as_deref().unwrap_or()` a common pattern that we would like to document?

Reply:
I guess not, that's up to the user to choose what they want to do if the directory is not a dSYM. It's like `path.parent()`.